core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -379,43 +379,49 @@ public void run() {
     }
   };
 
-  ManagedChannelImpl(String target, BackoffPolicy.Provider backoffPolicyProvider,
-      NameResolver.Factory nameResolverFactory, Attributes nameResolverParams,
-      LoadBalancer.Factory loadBalancerFactory, ClientTransportFactory transportFactory,
-      DecompressorRegistry decompressorRegistry, CompressorRegistry compressorRegistry,
+  ManagedChannelImpl(
+      AbstractManagedChannelImplBuilder<?> builder,
+      ClientTransportFactory clientTransportFactory,
+      BackoffPolicy.Provider backoffPolicyProvider,
       ObjectPool<ScheduledExecutorService> timerServicePool,
-      ObjectPool<? extends Executor> executorPool, ObjectPool<? extends Executor> oobExecutorPool,
-      Supplier<Stopwatch> stopwatchSupplier, long idleTimeoutMillis,
-      @Nullable String userAgent,
+      ObjectPool<? extends Executor> oobExecutorPool,
+      Supplier<Stopwatch> stopwatchSupplier,
       List<ClientInterceptor> interceptors) {
-    this.target = checkNotNull(target, "target");
-    this.nameResolverFactory = checkNotNull(nameResolverFactory, "nameResolverFactory");
-    this.nameResolverParams = checkNotNull(nameResolverParams, "nameResolverParams");
+    this.target = checkNotNull(builder.target, "target");
+    NameResolver.Factory tmpNameResolverFactory = builder.nameResolverFactory;
+    if (builder.authorityOverride != null) {
+      tmpNameResolverFactory = new OverrideAuthorityNameResolverFactory(
+          tmpNameResolverFactory, builder.authorityOverride);
+    }
+    this.nameResolverFactory = tmpNameResolverFactory;
+    this.nameResolverParams = checkNotNull(builder.getNameResolverParams(), "nameResolverParams");
     this.nameResolver = getNameResolver(target, nameResolverFactory, nameResolverParams);
-    this.loadBalancerFactory = checkNotNull(loadBalancerFactory, "loadBalancerFactory");
-    this.executorPool = checkNotNull(executorPool, "executorPool");
+    this.loadBalancerFactory =
+        checkNotNull(builder.loadBalancerFactory, "loadBalancerFactory");
```

Comment:
Without a nullability checker, I feel like they still serve a purpose. It seems like it would be pretty easy to change the builder without investigating whether this class supported null values.